src/g3log/logmessage.hpp

```diff
@@ -87,7 +88,8 @@ namespace g3 {
       // Complete access to the raw data in case the helper functions above
       // are not enough.
       //
-      std::time_t _timestamp;
+      // TODO: unique_ptr should be considered instead of shared_ptr
+      std::shared_ptr<Timestamp> _timestamp;
```

Comment:
Hi

I committed the first draft version based on the approach we discussed with some deviations (bare with me...)

- Instead of std::tm-like struct I kept existing std::time_t so default timestamp initialization remained the same; the reason is to minimize timestamp capture effort, because in case of tm-like struct there is a need to extra conversion std::time_t -> tm struct; besides, keeping std::time_t makes timestamp kind of compatible with chrono
- Then I introduced microsec field in timestamp struct which I plan to replace with boost approach - using num_fractional_digits and fractional_seconds fields instead
- There are two constructors in timestamps: a default one using g3::systemtime_now and a new one which allows providing microsec field (to be changed to num_fractional_digits and fractional_seconds)

It works and I can provide a custom function/functor/lambda to generate timestamps with fractional part (C++14 syntax):
	g3::setTimestampGenerator([format = std::string("%Y/%m/%d %H:%M:%S")]() mutable {
		auto ptime = boost::posix_time::microsec_clock::local_time();

		return g3::Timestamp(
			boost::posix_time::to_time_t(ptime),
			ptime.time_of_day().fractional_seconds(),
			format);
	});

Now I need to figure out formatting part. The straight forward way would be changing localtime_formatted's signature to accept timestamp object instead of std::time_t

Do you agree so far?